source/dofs/dof_tools_constraints.cc

```diff
@@ -1856,365 +1859,249 @@ namespace DoFTools
-                // be careful about treating identity constraints differently.
-                // consequently, find out whether this dof 'i' will be identity
-                // constrained
-                //
-                // to check whether this is the case, first see whether there
-                // are any weights other than 0 and 1, then in a first stage
-                // make sure that if so there is only one weight equal to 1
-                //
-                // afterwards do the same for constraints of type dof1=-dof2
-                bool         is_identity_constrained = true;
-                const double eps                     = 1.e-13;
-                for (unsigned int jj = 0; jj < dofs_per_face; ++jj)
-                  if (std::abs(transformation(i, jj)) > eps &&
-                      std::abs(transformation(i, jj) - 1.) > eps)
-                    {
-                      is_identity_constrained = false;
-                      break;
-                    }
-                unsigned int identity_constraint_target =
-                  numbers::invalid_unsigned_int;
-                if (is_identity_constrained == true)
+                const auto entry = transformation(i, jj);
+
+                if (std::abs(entry) > eps)
                   {
-                    bool one_identity_found = false;
-                    for (unsigned int jj = 0; jj < dofs_per_face; ++jj)
-                      if (std::abs(transformation(i, jj) - 1.) < eps)
-                        {
-                          if (one_identity_found == false)
-                            {
-                              one_identity_found         = true;
-                              identity_constraint_target = jj;
-                            }
-                          else
-                            {
-                              is_identity_constrained = false;
-                              identity_constraint_target =
-                                numbers::invalid_unsigned_int;
-                              break;
-                            }
-                        }
+                    no_entries++;
+                    is_identity_constrained = true;
+                    target                  = jj;
+                    factor                  = entry;
                   }
 
-                bool         is_inverse_constrained = !is_identity_constrained;
-                unsigned int inverse_constraint_target =
-                  numbers::invalid_unsigned_int;
-                if (is_inverse_constrained)
-                  for (unsigned int jj = 0; jj < dofs_per_face; ++jj)
-                    if (std::abs(transformation(i, jj)) > eps &&
-                        std::abs(transformation(i, jj) + 1.) > eps)
-                      {
-                        is_inverse_constrained = false;
-                        break;
-                      }
-                if (is_inverse_constrained)
+                if (no_entries > 1)
```

Comment:
I would prefer to avoid `no_entries` and merge the body of the `if`-switch into the one above so that it becomes
```
if (std::abs(entry) > eps)
  {
    // there is at least one more constraining dof
    if (is_identity_constrained)
      continue;
    is_identity_constrained = true;
    target                  = jj;
    factor                  = entry;
  }
```

Reply:
Good point.